management/commands/import_plos2EA.py

```diff
@@ -0,0 +1,495 @@
+
+    def parsename(self, namenode):
+        print("parse name")
+        #make sure name style is western
+        assert namenode.attrib['name-style']=="western", "expecting western stye name"
+        self.lname = namenode.find('surname').text
+        self.fname = namenode.find('given-names').text
+
+
+    def parseaddress(self, addrnode):
+        print("parse name")
+        self.email = addrnode.find('email').text
+        self.affil = addrnode.find('institution').text
+
+    def parseid(self, idnode):
+        print("parse id")
+        if idnode is not None and 'contrib-id-type' in idnode.attrib and idnode.attrib['contrib-id-type'] == "orcid":
+            print("FOUND ORCID")
+            self.orcid = idnode.text
+
+
+##############################
+class ArticleMetadata:
+    authors = []
+    subjects = []
+    keywords = []
+    customfields = []
+    title = None
+    abstract = None
+    reuse = None
+    sourceinfo = None
+    def __init__(self, root, sinfo):
+        print("created article metadata")
+        self.sourceinfo = sinfo
+        self.authors = []
+        self.subjects = []
+        self.keywords = []
+        self.customfields = []
+        self.fillData(root)
+        self.verifyData(root)
+
+    def fillData(self, root):
+        print("fill data for article")
+        self.title = root.find('front/article-meta/title-group/article-title').text
+        self.abstract = root.find('front/article-meta/abstract/p').text
+        contribsroot = root.find('front/article-meta/contrib-group')
+        for child in contribsroot:
+            #skip collab authors
+            if child.find('collab') is None:
+                self.authors.append(AuthorMetadata(child))
+        subjsroot = root.find('front/article-meta/article-categories/subj-group')
+        for child in subjsroot:
+            self.subjects.append(child.text)
+        keyroot = root.find('front/article-meta/kwd-group')
+        for child in keyroot:
+            self.keywords.append(child.text)
+        customroot = root.find('front/article-meta/custom-meta-group')
+        for child in customroot:
+            self.customfields.append(CustomMetadata(child))
+        self.reuse = root.find('front/article-meta/permissions/license/license-p').text
+
+    def verifyData(self, root):
+        print("verify the data")
+
+        #name sure required custom meta fields are present
+        approval = list(filter(lambda x: x.name == "author_approval", self.customfields))[0]
+        assert approval.value == 'true', "author approval is expected"
+        #make sure cc-by is present
+        assert self.reuse == "cc_by", "expecting cc_by"
+
+        #make sure there is atleast one corresponding author 
+        owner = list(filter(lambda x: x.corresp == "yes", self.authors))[0]
+        assert owner.email != None, "Owner is present"
+
+################################
+###### PreprintItem    #########
+###### Creates Objects #########

```

Comment:
Nice! I like terse documentation like this

Reply:
Thanks!